### PR TITLE
Reduce Task<T>/continuation usage in GrainReference/GrainFactory

### PR DESCRIPTION
**Pull request by ReubenBond**
Minor cleanup to reduce some redundant Task usage.

### CODE REVIEW

**Comment by gabikliot** [comment]
I am afraid there might be some backward compatibility issue. Maybe not. I need to double check.

File: src/Orleans/Runtime/GrainReference.cs [hunk 1]
```diff
@@ -136,9 +136,9 @@ internal static GrainReference NewObserverGrainReference(GrainId grainId, GuidId
         /// <summary>
         /// Called from generated code.
         /// </summary>
-        public static Task<GrainReference> CreateObjectReference(IAddressable o, IGrainMethodInvoker invoker)
+        public static GrainReference CreateObjectReference(IAddressable o, IGrainMethodInvoker invoker)
```
Comment (gabikliot) [inline src/Orleans/Runtime/GrainReference.cs]: I think you can get rid of this method completely and just call `RuntimeClient.Current.CreateObjectReference(o, invoker)` directly where this was called before. It was probably forgotten in some refactoring.

File: src/Orleans/Core/GrainFactory.cs [hunk 2]
```diff
@@ -158,16 +158,16 @@ public TGrainInterface GetGrain<TGrainInterface>(long primaryKey, string keyExte
         public Task<TGrainObserverInterface> CreateObjectReference<TGrainObserverInterface>(IGrainObserver obj)
```
Comment (gabikliot) [inline src/Orleans/Core/GrainFactory.cs]: You can kept that public method async, and that was the one I was worried about backward compat. The other method is internal, so no issues.

File: src/Orleans/Runtime/GrainReference.cs (hunk 1 above)
Comment (ReubenBond) [inline src/Orleans/Runtime/GrainReference.cs]: will do

**Pull request merged**